### PR TITLE
Use 'xnethackdir' on macOS, add install script

### PR DESCRIPTION
**Pull request by miguno**
As per the discussions at https://github.com/copperwater/xNetHack/issues/62.

Notes:
- I picked `$Top/install_macos.sh` as the name and location for the install script. Reasoning: Should be easy to find (as a beginner, I would look for such a script in the top-level directory, not several directories deep under, e.g., `sys/unix/...`), and there was no `$Top/contrib/` directory yet. Let me know if you prefer something else, such as under `sys/unix/`. I can only give you the beginner POV, not the POV of NetHack veterans that would already know where to find the tooling for their respective operating systems. :)
- The changes made to `sys/unix/hints/{macOS.370,macosx.sh)` where minimal, even though there are still several leftover `NetHack` references. I focused on HACKDIR-related changes. Many others are referring to NetHack code files, which as far as I can tell were not renamed in xNetHack to minimize merge conflicts from upstream NetHack.

### CODE REVIEW

**Comment by miguno** [comment]
The install script could also be mentioned in the top-level [README.md](https://github.com/copperwater/xNetHack/blob/master/README.md).

File: install_macos.sh [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+echo "About to install xNetHack:"
+echo
+echo "- Install directory: $DEFAULT_HACKDIR"
+echo "- Launch script:     $LAUNCH_SCRIPT"
+echo
+echo "***                   WARNING WARNING WARNING                    ***"
+echo "***  Backup your existing install directory before proceeding,   ***"
+echo "***  because it will be OVERWRITTEN by the new installation.     ***"
+echo "***  Without a backup, you lose your record, logfile, sysconf    ***"
+echo "***  files, etc.                                                 ***"
```
Comment (copperwater) [inline install_macos.sh]: Also stick save files in this list, since most single-user installs will primarily care about them :)

**Review by copperwater — COMMENTED** [review]
The NetHack veteran POV (or possibly a DevTeam POV) would be that it's bad precedent to put install stuff for one platform at the top level, because then why doesn't everything else get top level install stuff too? 

I guess the proper approach would be to put it in sys/unix and mention it in README.md that mac users can run the install_macos.sh script to get the basic install with multiple windowport builds.

File: install_macos.sh (hunk 1 above)
Comment (miguno) [reply install_macos.sh]: Good point, added!

**Comment by miguno** [comment]
> I guess the proper approach would be to put it in sys/unix and mention it in README.md that mac users can run the install_macos.sh script to get the basic install with multiple windowport builds.

Updated PR accordingly.

**Comment by miguno** [comment]
> [...] because then why doesn't everything else get top level install stuff too?

To be fair, in an ideal world, I would have expected (a) ready-to-use release binaries for, say, Windows, macOS, Linux under https://github.com/copperwater/xNetHack/releases, given that there is already a [cross-compilation workflow in place](https://github.com/copperwater/xNetHack/blob/master/.github/workflows/xnethack-cross-platform-build.yml) anyway, or at least (b) one-click build+install tooling for the major platforms, to help onboarding new users/developers more quickly.

(Same can be said for NetHack.)

No critique, just general feedback since you brought it up. The bar of expectations is higher nowadays, given all the other cool projects on GitHub that are doing the very thing, which means users are spoiled. ;-)

**Comment by miguno** [comment]
Does the latest update address your comments, @copperwater ?

**Pull request merged**